articles/flask-database-integration-with-sqlalchemy/index.md

```diff
@@ -0,0 +1,395 @@
+
+```python
+@app.route("/register", methods=["GET", "POST"])
+def register():
+    # check the request method to ensure the handling of POST request only
+    if request.method == "POST":
+        # store the form value
+        user_name = request.form["username"]
+        email = request.form["email"]
+        password = request.form["password"]
+        
+        # create an instance of the user table
+        user = User(username = user_name, user_email = email, user_password = password)
+        
+        return username + " <br/> " + email
+    return render_template('register.html')
+```
+Next, we add the user and commit the changes to the database and we replace the return statement with “User registration successful”. By the end of the implementation, your registration route should like this.
+
+```python
+
+@app.route("/register", methods=["GET", "POST"])
+def register():
+    #check the request method to ensure the handling of POST request only
+    if request.method == "POST":
+        #store the form value
+        user_name = request.form["username"]
+        email = request.form["email"]
+        password = request.form["password"]
+        
+        # store the user details in the user database table
+        user = User(username = user_name, user_email = email, user_password = password)
+        
+        # add the user object to the database
+        db.session.add(user)
+        
+        # commit changes to the database 
+        db.session.commit()
+        
+        return 'User registration successful'
+
+    return render_template('register.html')
+
+```
+
+### Test Running Our App
+Let’s test run our app. If you have the server running, terminate and restart. Now, you should have your server running, then proceed to the registration page on your web app. Let’s fill the data with this dummy data: username = Peter, email = peterdury@gmail.com, password= password.
+
+![Registration successful](/engineering-education/flask-database-integration-with-sqlalchemy/response.png)
+
+
+We received the registration successful message, but let’s check the database to check if our user data is stored.
+
+```bash
+# import db and User table
+>>> from app import db
+>>> from app.models import User
+
+# query User table
+>>> User.query.all()
+[<User 1>, <User 2>]
+```
+When you run the following command in sequential order, you should get a response indicating the new User added. In our case, we add a user when we added one manually, and due to the registration, we now have a new user.
+
+Let’s delve deeper and run a new command to query the User table by the username of the registered user.
+
+```bash
+# store queried data in the user_data variable
+>>>> user_data = User.query.all()
+# loop through the user data
+>>>> for user in user_data:
+      # print user username in the User table
+….    print(user.username)
```

Comment:
```suggestion
>>>> for user in user_data:
      # print user username in the User table
      print(user.username)
```

Reply:
Also, note here that you have used `>>>>` which is inconsistent with the previous one `>>>`.

Make it consistent throughout the article.

Reply:
@corpsgeek